### PR TITLE
Upgrade Bytes phase 2

### PR DESCRIPTION
**Pull request by holisticode**

### CODE REVIEW

File: pkg/subnet/upgrades/upgrade_file.go [hunk 1]
```diff
@@ -14,19 +13,34 @@ import (
 	"github.com/ava-labs/avalanchego/utils/storage"
 )
 
+func WriteLockUpgradeFile(jsonBytes []byte, subnetName string, app *application.Avalanche) error {
```
Comment (dasconnor) [inline pkg/subnet/upgrades/upgrade_file.go:16]: We manage all other file operations through the `application` package

Comment (dasconnor) [reply pkg/subnet/upgrades/upgrade_file.go:16]: Also, it's very unclear to me what the contents of the lock file actually are. I suggest creating a lock file type (or reusing an existing type) and accepting that as an argument, then do any type conversions in this method, then finally call out to the app package to write the file.

File: pkg/subnet/upgrades/upgrade_file.go [hunk 2]
```diff
@@ -37,25 +51,31 @@ func WriteUpgradeFile(jsonBytes []byte, subnetName string, app *application.Aval
...
 }
 
 func ReadUpgradeFile(subnetName string, app *application.Avalanche) ([]byte, error) {
 	localUpgradeBytesFilePath := app.GetUpgradeBytesFilePath(subnetName)
 
-	exists, err := storage.FileExists(localUpgradeBytesFilePath)
+	return readFile(localUpgradeBytesFilePath)
+}
+
+func ReadLockUpgradeFile(subnetName string, app *application.Avalanche) ([]byte, error) {
```
Comment (dasconnor) [inline pkg/subnet/upgrades/upgrade_file.go:63]: Same comment as above. Use the app package to read the file, do type conversions and return a mature data type, not []byte

**Review by dasconnor — COMMENTED** [review]
Just need to improve some of the file IO, then I think it'll be good

File: pkg/subnet/upgrades/upgrade_file.go (hunk 1 above)
Comment (holisticode) [reply pkg/subnet/upgrades/upgrade_file.go]: OK moving to `application` package__

**Review by dasconnor — APPROVED** [review]
Looks a lot cleaner!

**Pull request merged**